test/integration/action_client/active_job_assertions_test.rb

```diff
@@ -0,0 +1,226 @@
+require "test_helper"
+require "active_job_test_case"
+
+module ActionClient
+  class ActiveJobAssertionsTestCase < ActiveJobTestCase
+    include ActionClient::TestHelpers
+    MetricsClient = Class.new(ActionClient::Base) {
```

Comment:
Are these different if you are assigning the class to a constant?

```suggestion
    class MetricsClient < ActionClient::Base
```

Reply:
From a Ruby perspective, they're interchangeable.

I've tried to establish a convention of using `Class.new` in tests to signal that the class being declared is "test data". I've found it a good visual cue to separate them from TestCase classes (since MiniTest blocks use the `class` keyword).

I'm not sure on how effective it's actually been. I'm the only one doing active development, so I'm biased. If you find it confusing, then that's a good indicator that it isn't helpful!

Reply:
Is it possible to assign this to a local variable instead of a constant so it can get cleaned up after the test is finished?

Reply:
We do that elsewhere through the `declare_client` test helper:

https://github.com/thoughtbot/action_client/blob/a7fb5e979cc3b73ac57073fafdd27cff0b12f1a2/test/integration_test_case.rb#L8-L20

That usually works to create anonymous classes (that are sometimes named for view path resolution).

Unfortunately, the ActiveJob classes need to be resolvable by their name, since they're serialized and deserialized to the job queue. That constraint has made it more difficult to use test block-scoped anonymous classes. 

Do you have something particularly different in mind?

Reply:
That makes sense. I think what you have is fine.

In factory_bot we have a [test helper for defining constants](https://github.com/thoughtbot/factory_bot/blob/master/spec/support/macros/define_constant.rb#L4-L8) that uses [RSpec's `stub_const`](https://relishapp.com/rspec/rspec-mocks/docs/mutating-constants). I don't know if there is an equivalent here.

Reply:
There's a MiniTest gem for [stub_const](https://github.com/adammck/minitest-stub-const).

I think combining that with the `define_class` helper could be pretty sweet!

https://github.com/thoughtbot/factory_bot/blob/5c071d42fda1682282461978923a2bd89f2ad27d/spec/support/macros/define_constant.rb#L4-L8